src/k8s/pkg/k8sd/types/cluster_config_certificates.go

```diff
@@ -21,6 +23,14 @@ func (c Certificates) GetAPIServerKubeletClientCert() string {
 func (c Certificates) GetAPIServerKubeletClientKey() string {
 	return getField(c.APIServerKubeletClientKey)
 }
+
+// Check if every field of the Certificate struct is Nil, returns false otherwise.
 func (c Certificates) Empty() bool {
-	return c.CACert == nil && c.CAKey == nil && c.FrontProxyCACert == nil && c.FrontProxyCAKey == nil && c.ServiceAccountKey == nil && c.APIServerKubeletClientCert == nil && c.APIServerKubeletClientKey == nil
+	val := reflect.ValueOf(c)
+	for i := 0; i < val.NumField(); i++ {
+		if val.Field(i).Kind() == reflect.Pointer && !val.Field(i).IsNil() {
+			return false
+		}
+	}
+	return true
```

Comment:
not a huge fan of this. i'd rather we instead have a unit test that uses reflect to set individual fields (and would fail in case `.Empty()` does not return true.

If you feel like this grows a lot, then consider rewriting as:

```
switch {
case c.CACert != nil:
  return false
case c.CAKey != nil:
  return false
// ...
}
return true
}
```

Reply:
+1 That will help with readability as well.